src/main/java/io/debezium/connector/db2/Db2StreamingChangeEventSource.java

```diff
@@ -292,19 +293,26 @@ else if (operation == Db2ChangeRecordEmitter.OP_UPDATE_SINGLE) {
                     tablesSlot.set(processErrorFromChangeTableQuery(e, tablesSlot.get()));
                 }
 
-                if (context.isPaused()) {
-                    LOGGER.info("Streaming will now pause");
-                    context.streamingPaused();
-                    context.waitSnapshotCompletion();
-                    LOGGER.info("Streaming resumed");
-                }
+                handlePause(context);
             }
         }
         catch (Exception e) {
             errorHandler.setProducerThrowable(e);
         }
     }
 
+    private void handlePause(final ChangeEventSourceContext context){
+        try{
+            if (context.isPaused()) {
+                LOGGER.info("Streaming will now pause");
+                context.streamingPaused();
+                context.waitSnapshotCompletion();
+                LOGGER.info("Streaming resumed");
+            }
+        }catch (Exception e) {
+            errorHandler.setProducerThrowable(e);
+        }
+    }
```

Comment:
Could we not let the error bubble up and let the catch block in the execute method handle setting the exception on the producer?